### PR TITLE
Add `MsgFlag::MSG_NOSIGNAL`

### PR DESCRIPTION
**Pull request by i509VCB**
(I guess I will see if I did the cfg block correctly)

### CODE REVIEW

**Comment by asomers** [comment]
Looks good, but could you please add a CHANGELOG entry?

**Comment by i509VCB** [comment]
Changelog is done, I checked libc and realized I forgot illumos so that is added.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 ## [Unreleased] - ReleaseDate
 ### Added
 
+- Added `MSG_NOSIGNAL` for Android, Dragonfly, FreeBSD, Fuchsia, Haiku, Illumos, Linux, NetBSD, OpenBSD and Solaris.
```
Comment (asomers) [inline CHANGELOG.md:9]: Please follow the format of the other changelog entries.

**Review by asomers — CHANGES_REQUESTED** [review]

**Pull request merged**